zeroex/order.go

```diff
@@ -154,6 +171,25 @@ func (s *SignedOrder) ComputeOrderHash() (common.Hash, error) {
 	return hash, nil
 }
 
+// ECSign signs the order with it's `makerAddress` and formats it to be a valid 0x order signature
+func (s *SignedOrder) ECSign(rpcClient *rpc.Client) ([]byte, error) {
```

Comment:
Can we make a separate `Order` type to compliment the `SignedOrder` type? Then we could make `ECSign` a method of `Order` or a function that takes an `Order` as an argument. I think it would be more clear and would also allow us to catch certain bugs at compile time. The way it is currently written, passing a `SignedOrder` with a `nil` signature to functions which expect a signature can result in a runtime error.